forge-gui/res/cardsfolder/upcoming/gollum_scheming_guide.txt

```diff
@@ -0,0 +1,19 @@
+Name:Gollum, Scheming Guide
+ManaCost:1 B
+Types:Legendary Creature Halfling Horror
+PT:2/1
+T:Mode$ Attacks | ValidCard$ Card.Self | Execute$ TrigPeek | TriggerZones$ Battlefield | TriggerDescription$ Whenever CARDNAME attacks, look at the top two cards of your library, put them back in any order, then choose land or nonland. An opponent guesses whether the top card of your library is the chosen kind. Reveal that card. If they guessed right, remove NICKNAME from combat. Otherwise, you draw a card and NICKNAME can't be blocked this turn.
+SVar:TrigPeek:DB$ RearrangeTopOfLibrary | Defined$ You | NumCards$ 2 | SubAbility$ DBChoosePlayer
+SVar:DBChoosePlayer:DB$ ChoosePlayer | Defined$ You | Choices$ Opponent | ChoiceTitle$ Choose an opponent | SubAbility$ DBChoose
+SVar:DBChoose:DB$ ChooseType | Type$ Card | ValidTypes$ Land,Nonland | Defined$ You | SubAbility$ DBChooseBis
+SVar:DBChooseBis:DB$ ChooseType | Defined$ ChosenPlayer | Type$ Card | ValidTypes$ Land,Nonland | SubAbility$ DBReveal
+SVar:DBReveal:DB$ PeekandReveal | PeekAmount$ 1 | RevealValid$ Card.ChosenType | RememberRevealed$ True | SubAbility$ DBBranch
```

Comment:
Needs to reveal the Card either way... better to just remove the RevealValid here because the default will reveal it
But needs to remember only if it's the Chosen Type
What it needs is a `RememberValid$ Card.ChosenType` but we'd need to wire up `RememberValid`

Reply:
Imo the param can just be removed and then on the branch effect you can check for the type via the SVar?